File: backend/app/schemas/interview.py
"""
Echo Mock System - 面试场次相关 Pydantic 模型
"""

from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


# =============================================
# 请求模型 (前端 -> 后端)
# =============================================

class InterviewCreateRequest(BaseModel):
    target_role: str = Field(..., description="目标岗位（如 Java 后端、前端）")


# =============================================
# 响应模型 (后端 -> 前端)
# =============================================

class InterviewBrief(BaseModel):
    """面试场次摘要 - 用于列表展示"""
    id: str
    target_role: str
    status: str
    start_time: datetime
    end_time: Optional[datetime] = None
    overall_score: Optional[float] = None

    class Config:
        from_attributes = True


class InterviewDetail(BaseModel):
    """面试场次详情 - 包含完整对话与评分"""
    id: str
    target_role: str
    status: str
    start_time: datetime
    end_time: Optional[datetime] = None
    overall_score: Optional[float] = None
    dimension_scores: Optional[Dict[str, Any]] = None
    comprehensive_report: Optional[str] = None
    messages: List["MessageItem"] = []

    class Config:
        from_attributes = True


class MessageItem(BaseModel):
    """单条对话消息"""
    id: str
    round_seq: int
    speaker: str
    content: str
    audio_file_path: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/interview.ts
/**
 * Echo Mock System - 面试场次 API 请求
 */

import request from '../utils/request'

export interface InterviewCreateRequest {
  target_role: string
}

export interface InterviewBrief {
  id: string
  target_role: string
  status: string
  start_time: string
  end_time: string | null
  overall_score: number | null
}

export interface MessageItem {
  id: string
  round_seq: number
  speaker: string
  content: string
  audio_file_path: string | null
  created_at: string
}

export interface InterviewDetail extends InterviewBrief {
  dimension_scores: Record<string, any> | null
  comprehensive_report: string | null
  messages: MessageItem[]
}

/** 创建面试场次 */
export const createInterview = (data: InterviewCreateRequest) =>
  request.post<any, InterviewBrief>('/interviews/', data)

/** 获取面试历史列表 */
export const listInterviews = () =>
  request.get<any, InterviewBrief[]>('/interviews/')

/** 获取面试场次详情 */
export const getInterviewDetail = (sessionId: string) =>
  request.get<any, InterviewDetail>(`/interviews/${sessionId}`)
